### ISSUE TITLE
Indentation of public, protected, private keywords

### ISSUE THREAD

**Issue by thomasgeissl** [openframeworks/openFrameworks#4448]
I am suggesting a little change to the coding style used for oF. 
I would like to remove the indentation of public, protected and private keywords.

Here is the snippet from the style guidelines I am referring to:
The ```public```, ```protected```, and ```private``` keywords should be indented inside the class with the function declarations indented as well.

Every time a new line is added in a header file, it gets indented according to the IDE's (in my case Xcode) default style. And one has to manually indent the line.

![screen shot 2015-10-18 at 12 41 15](https://cloud.githubusercontent.com/assets/3796674/10563719/9a6deae8-7595-11e5-98ed-e40e8dc852ac.png)

I guess most people using Xcode as their IDE (I haven't checked the latest survey) and the default Xcode style is the default for other IDEs too, as Arturo says in (https://github.com/openframeworks/projectGenerator/issues/59). I can confirm Qt Creator uses the same styling.

In my opinion this often leads to bad formatted code.

Another point I would like to change is the order of the following two lines in the ofApp template. They should be swapped to be in the same order as in the implementation file. 

```c++
void dragEvent(ofDragInfo dragInfo);
void gotMessage(ofMessage msg);
```

What were the reasons for the indentation? And what are the disadvantages of changing the style guidelines?

**Comment by danoli3** [comment]
Also Xcode makes it easy to make this change if this is the path we go down.
Select All - > Cut ->Paste 

Easy to Xcode Default indentation :)

**Comment by julapy** [comment]
yeah +1
that always niggles at me too.
i also do what @danoli3 does, cut and paste to get it back into xcode format.
be nice if it was already formatted correctly tho...

**Comment by flvi0** [comment]
dudes.... get to know your tools.
https://www.gnu.org/software/indent/
brew install gnu-indent

here's PR: https://github.com/openframeworks/openFrameworks/pull/4453

Created with 
```
find . -name '*.cpp' -o -name '*.h' -exec indent -ci4 -i4 -nut {} \;
```

You might want to take this opportunity to have your tiresome discussion about where braces need to be placed, etc. however I think there are more pressing issues at hand ;)

**Comment by bilderbuchi** [comment]
dude, get to know _your_ tools: https://github.com/openframeworks/openFrameworks/wiki/openFrameworks-Coding-style-guidelines and the /scripts/dev/style directory.
also, time to get off your high horse I think. ;-)

**Comment by flvi0** [comment]
Ah nice so there is stuff available regarding this. Well don't have an overview about all the details of the project yet, scusi :/ Was just irritated that copy/paste is used for reformatting of sourcecode. And yeah maybe I should step back here a little bit, thanks for pointing that out. Anyways I have seen that there is no consistently applied style right now by looking at the different headers, so I think having this automated solution which will indent the complete source tree for once would be nice.

(wtf happened with that const keyword? looking into it...)

**Comment by bilderbuchi** [comment]
If you look at /scripts/dev/style, you will find that an automated solution (using `uncrustify`) already exists, although it probably needs a bit of updating.  
The reason we haven't formatted OF with this yet is that it was felt by @openframeworks/core that the whitespace changes would pollute the git history; and because there was not mechanism in place to ensure compliance with the style after we applied it - if that's not there, the code would slowly go "out of style" again. In the meantime, we got a CI setup using Travis, so maybe it's time to revisit the question of an automated checker (based on styleWholeOF).
There [seems to be](https://github.com/bengardner/uncrustify/blob/fc5228ecd9f579f366919e5d13f26ae5149f80f1/src/options.h#L143-L144) an uncrustify setting doing what you want, so you can probably play around with that in the style config file to see if that matches your expectations.

see also issue #3392 and probably some threads on the mailing list (if you want earlier history).

**Comment by danoli3** [comment]
Yeah it really would break a lot of old PR's that are yet to be looked at. 
If this is a thing to do, have to really take that into consideration.
